src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -361,13 +361,14 @@ record = _serverStore.Cluster.ReadDatabase(context, Name, out index);
                         RachisLogIndexNotifications.NotifyListenersAbout(index, e);
                     }
                 }, null);
-
-                _clusterTransactionsTask = Task.Run(async () =>
+                var threadName = $"{Name} Cluster Transaction Thread";
+                _clusterTransactionsThread = PoolOfThreads.GlobalRavenThreadPool.LongRunning(x =>
                 {
+                    ThreadHelper.TrySetThreadPriority(ThreadPriority.AboveNormal, threadName, _logger);
```

Comment:
I don't think that setting `ThreadPriority.AboveNormal` here is needed or recommended, unless you prove to me otherwise :-)

I see that we have this set in the TxMerger and in the rachis threads (expect for the Leader??)

I can argue that setting higher priority for the cluster threads is essential, to maintain a stable cluster over transactions throughput. 

Maybe we can set `ThreadPriority.AboveNormal` for this and TxMerger but `ThreadPriority.Highest` for cluster threads?
cc @ayende thoughts?

Reply:
I would expect this to be high, yes. Because that may be blocking requests that are waiting for it to complete.
The leader *should* be `AboveNormal` as well

In terms of priorities:

* Cluster (leader, follower, etc)
* Tx Mergers for databases
* Requests


Not sure whatever we actually want to use more than two levels, though.